Build warning: 'row_names' uninitialized

Happened to be building on Windows and saw this. I'll fix this tomorrow. 

```
slice.c: In function 'split_along_shaped':
slice.c:686:23: warning: 'row_names' may be used uninitialized in this function [-Wmaybe-uninitialized]
         new_row_names = slice_names(row_names, index);
```

Can probably be fixed by setting `row_names = NULL` here
https://github.com/r-lib/vctrs/blob/f7d2492cdf3223a62d5023604a5652d29555fd76/src/slice.c#L662

--- a/src/slice.c
+++ b/src/slice.c
@@ -659,7 +659,7 @@ SEXP split_along_shaped(SEXP x, struct vctrs_proxy_info info) {
 
   SEXP dim_names = PROTECT(Rf_getAttrib(x, R_DimNamesSymbol));
 
-  SEXP row_names;
+  SEXP row_names = R_NilValue;
   if (dim_names != R_NilValue) {
     row_names = VECTOR_ELT(dim_names, 0);
   }

